python/ray/serve/schema.py

```diff
@@ -63,6 +64,51 @@ def _route_prefix_format(cls, v):
     return v
 
 
+@PublicAPI(stability="alpha")
+class LoggingConfigSchema(BaseModel):
+    """Logging config schema for configuring serve components logs."""
+
+    encoding: str = Field(
+        default="TEXT",
+        description=(
+            "Encoding type for the serve logs. Default to 'TEXT'. 'JSON' is also "
+            "supported to format all serve logs into json structure."
+        ),
+    )
+    log_level: int = Field(
+        default=logging.INFO,
+        description=(
+            "Log level for the serve logs. Default to logging.INFO. You can set it to "
+            "'logging.DEBUG' to get more detailed logs."
+        ),
+    )
+    logs_dir: Union[str, None] = Field(
+        default=None,
+        description=(
+            "Directory to store the logs. Default to None, which means "
+            "logs will be stored in the default directory "
+            "('/tmp/ray/session_latest/logs/serve/...')."
+        ),
+    )
+    enable_access_log: bool = Field(
+        default=True,
+        description=(
+            "Whether to write serve log to the file. Default to True."
+            " When set to False, the serve logs are not written to files."
+        ),
+    )
+
+    @validator("encoding")
+    def valid_encoding_format(cls, v):
+
+        if v not in ["TEXT", "JSON"]:
```

Comment:
Kinda after thought, this can be more foolproof by doing `if v not in list(EncodingType)` 🙃

Reply:
done!